pkg/destroy/aws/aws.go

```diff
@@ -82,6 +86,20 @@ func New(logger logrus.FieldLogger, metadata *types.ClusterMetadata) (providers.
 		return nil, err
 	}
 
+	cfg, err := configv2.LoadDefaultConfig(context.TODO(), configv2.WithRegion(region))
+	if err != nil {
+		return nil, fmt.Errorf("failed loading default config: %w", err)
+	}
+	ec2Client := ec2v2.NewFromConfig(cfg, func(options *ec2v2.Options) {
+		options.Region = region
+
+		for _, endpoint := range metadata.AWS.ServiceEndpoints {
+			if strings.EqualFold(endpoint.Name, "ec2") {
+				options.BaseEndpoint = aws.String(endpoint.URL)
```

Comment:
Nit: Do we expect only one ec2 endpoint? If yes, we could break here. If not, is options.BaseEndpoint a list (or comma separated string) of base endpoints?